org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java

```diff
@@ -360,13 +366,26 @@ private void doSubmitModel(SModelRoot newRoot, boolean update, Action cause) {
 					dispatch(new SetModelAction(newRoot));
 				}
 				lastSubmittedModelType = modelType;
+				updateSelection(newRoot, update);
 				IModelUpdateListener listener = getModelUpdateListener();
 				if (listener != null) {
 					listener.modelSubmitted(newRoot, this);
 				}
 			}
 		}
 	}
+
+	private void updateSelection(SModelRoot newRoot, boolean update) {
+		if (update) {
+			if (!selectedElements.isEmpty()) {
+				Set<String> removedElements = Sets.newHashSet(selectedElements);
+				new SModelIndex(newRoot).allIds().forEach(removedElements::remove);
+				selectedElements.removeAll(removedElements);
```

Comment:
We could change the type of `allIds()` to `Set<String>` and then simplify this to
```
selectedElements.retainAll(new SModelIndex(newRoot).allIds());
```

Reply:
In that case even the `isEmpty()` check would not be necessary.

Reply:
That's a lot shorter indeed, updated!